### PR TITLE
added reverse replication limitations

### PR DESCRIPTION
**Pull request by aksharauke**
added reverse replication limitations

### CODE REVIEW

File: reverse_replication/README.md [hunk 1]
```diff
@@ -89,4 +89,33 @@ go run launcher.go -projectId=my-project -dataflowRegion=us-east1 -instanceId=my
...
+2. Adding column in Spanner that does not exist in source - in this case the column cannot be replicated
+3. Deleting column in Spanner that is mandatory in source
+4. Spanner PK is UUID while source PK is auto-increment key 
+5. Spanner table has more columns as part of PK than the source 
+6. Spanner columns have greater length of string columns than source 
+7. Spanner columns have different data type than source 
+8. CLOB will not be read from GCS and put in source 
+9. DELETES on Spanner that have Primary key columns different from the Source database column - such records will be dropped
+10. Primary key of the source table cannot be determined - such records will be dropped
+In the above cases, custom code will need to be written to perform reverse transformation
```
Comment (manitgupta) [inline reverse_replication/README.md:121]: Can we give the link to the reverse transformation Dataflow template that the user can fork to write custom code?

File: reverse_replication/README.md [hunk 2]
```diff
@@ -89,4 +89,33 @@ go run launcher.go -projectId=my-project -dataflowRegion=us-east1 -instanceId=my
...
+1. There is an unexpected spike on Spanner
+2. The pipeline is started with a date in past ( due to issues that required downtime such as bug fix )
+Currently, there is no way to revert this within the same job. More details and recovery steps below.
+
+### Stuck pipeline recovery 
+1. The user must track the last handled window that was successfully processed - this can be obtained from the Dataflow logs or by checking the watermark of the job stage.
+2. Specify a different partition metadata database or delete the previous metadata tables - since watermarks for a given partition are persisted and hence will be read from that point onwards. So the pipeline must begin with a clean partition metadata database.
+3. The current pipeline must be updated - the user must specify the start time of change stream query as the last successfully processed timestamp and the end time of the change stream as the time that would result in ~1GB writes. Example, if average record size is 1KB, then 10,00,000 records should be processed in a window and if the TPS during that window was 20K, then the window must end at 50 second
```
[the remark this answers is not in the document]
Comment (aksharauke) [reply reverse_replication/README.md:105]: Gave link to Dataflow documentation, it has details of not just the gcloud way but also programmatic way.

File: reverse_replication/README.md [hunk 3]
```diff
@@ -89,4 +89,33 @@ go run launcher.go -projectId=my-project -dataflowRegion=us-east1 -instanceId=my
...
+4. Once this window is processed, the pipeline must be updated with next set of start and end times, until the pipeline catches up and the finally the end timestamp need not be passed.
+
+
+## Reverse transformations
```
[the remark this answers is not in the document]
Comment (aksharauke) [reply reverse_replication/README.md:109]: This and the stuck pipeline are the two known limitation categories. The recovery steps is the sub-section of the stuck pipeline. Added a line to make it clearer. Do let me know if there is still a concern.

File: reverse_replication/README.md (hunk 1 above)
Comment (aksharauke) [reply reverse_replication/README.md:121]: Done

File: reverse_replication/README.md [hunk 4]
```diff
@@ -89,4 +89,33 @@ go run launcher.go -projectId=my-project -dataflowRegion=us-east1 -instanceId=my
...
+
+## Stuck pipeline from Spanner to Sink
+The pipeline that reads the change streams and writes to PubSub gets stuck retrying the same set of records when the stuck pipeline issue occurs. The scenario can arise when there are a lot of changestream records to be read in a short interval of time, which occurs in  the following situations:
+1. There is an unexpected spike on Spanner
+2. The pipeline is started with a date in past ( due to issues that required downtime such as bug fix )
+Currently, there is no way to revert this within the same job. More details and recovery steps below.
+
+### Stuck pipeline recovery 
+1. The user must track the last handled window that was successfully processed - this can be obtained from the Dataflow logs or by checking the watermark of the job stage.
+2. Specify a different partition metadata database or delete the previous metadata tables - since watermarks for a given partition are persisted and hence will be read from that point onwards. So the pipeline must begin with a clean partition metadata database.
```
[the remark this answers is not in the document]
Comment (aksharauke) [reply reverse_replication/README.md:104]: Done

File: reverse_replication/README.md [hunk 5]
```diff
@@ -89,4 +89,33 @@ go run launcher.go -projectId=my-project -dataflowRegion=us-east1 -instanceId=my
...
 ```
 go run launcher.go -projectId=my-project -dataflowRegion=us-east1 -instanceId=my-instance -dbName=mydb -sourceShardsFilePath=gs://bucket-name/shards.json  -sessionFilePath=gs://bucket-name/session.json -pubSubEndpoint=pubsub.googleapis.com:443
-```
\ No newline at end of file
+```
+
+# Reverse Replication Limitations
+
+## Stuck pipeline from Spanner to Sink
+The pipeline that reads the change streams and writes to PubSub gets stuck retrying the same set of records when the stuck pipeline issue occurs. The scenario can arise when there are a lot of changestream records to be read in a short interval of time, which occurs in  the following situations:
```
[the remark this answers is not in the document]
Comment (aksharauke) [reply reverse_replication/README.md:97]: Done

**Review by manitgupta — APPROVED** [review]

**Pull request merged**